controllers/directoryservice_controller.go

```diff
@@ -84,9 +84,16 @@ func (r *DirectoryServiceReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	var clusterIdentifier string = ""
 	svcName := ds.Name
 
+	// Set default address required for DS Proxy
+	if ds.Spec.ServerConfig.AdvertisedListenAddress == "" {
```

Comment:
you can add the extra condition here. This way `ds.Spec.ServerConfig.AdvertisedListenAddress` is only modified in one place. not a big deal but might help with readability. 

```suggestion
	if ds.Spec.ServerConfig.AdvertisedListenAddress == "" &&  ds.Spec.MultiCluster.ClusterTopology == ""{
```

Reply:
- [ ] Review condition

Reply:
Yeah I was a bit torn in this because there are 3 options:

- Default setting compatible with DS Proxy(as you previously configured and I removed to get MCS working)
- User provided value in custom resource
- MCS value

I wonder if we really need the user provided value?  Because the reason why I set it twice is it needs to be removed for MCS to ensure its dynamically set.   I might remove that as it doesn't then need to be documented as an extra option for the user as it doesn't update the spec.  Its just default(works with DS proxy) or empty if MCS config provided.

Reply:
yeah. I agree, just simplify. If not MCS then default to POD_NAME